styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -464,6 +470,56 @@ private void initializeResources(Storage storage, ShardedCounter shardedCounter,
       LOG.error("Error reading resources from datastore: {}", e);
       // TODO: re-throw exception when moving to entirely depend on counter shards
     }
+
+    try {
+      if (storage.config().resourcesSyncEnabled()) {
+        try {
+          final Map<String, Long> resourcesUsageMap = getResourcesUsageMap(storage, timeoutConfig,
+              workflowCache, time.get(), resourceDecorator);
+          updateShards(storage, resourcesUsageMap);
+        } catch (Exception e) {
+          LOG.error("Error syncing resources: {}", e);
+          // TODO: re-throw exception when moving to entirely depend on counter shards
+        }
+      }
+    } catch (IOException e) {
+      LOG.error("Couldn't retrieve configuration for resource syncing", e);
+      // TODO: re-throw exception when moving to entirely depend on counter shards
```

Comment:
I think we can re-throw this today already.

Reply:
+1